sdk/transaction.go

```diff
@@ -36,6 +32,101 @@ func (txs *TransactionService) GetTransaction(ctx context.Context, id string) (T
+	if err != nil {
+		return nil, err
+	}
+
+	if err = handleResponseStatusCode(resp, map[int]error{404: ErrResourceNotFound, 409: ErrArgumentNotValid}); err != nil {
+		return nil, err
+	}
+
+	return MapTransactions(&b, txs.client.GenerationHash())
+}
+
+// returns Transaction for passed transaction id or hash
+func (txs *TransactionService) GetTransaction(ctx context.Context, id string) (Transaction, error) {
+	trS, err := txs.GetTransactionStatus(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	return txs.getTransaction(ctx, trS.Group, id)
+}
+
+// returns confirmed Transaction by id or hash
+func (txs *TransactionService) GetConfirmedTransaction(ctx context.Context, id string) (Transaction, error) {
+	return txs.getTransaction(ctx, confirmed.String(), id)
+}
+
+// returns confirmed Transactions
+func (txs *TransactionService) GetConfirmedTransactions(ctx context.Context, tpOpts *TransactionsPageOptions) (*TransactionsPage, error) {
+	return txs.getTransactionsByGroup(ctx, confirmed, tpOpts)
+}
+
+// returns an array of Transaction's for passed array of transaction ids or hashes
+func (txs *TransactionService) GetConfirmedTransactionsByIds(ctx context.Context, ids []string) ([]Transaction, error) {
+	return txs.getTransactionsByIds(ctx, confirmed, ids)
+}
+
+// returns unconfirmed Transaction by id or hash
+func (txs *TransactionService) GetUnconfirmedTransaction(ctx context.Context, id string) (Transaction, error) {
+	return txs.getTransaction(ctx, unconfirmed.String(), id)
+}
+
+// returns unconfirmed Transactions
+func (txs *TransactionService) GetUnconfirmedTransactions(ctx context.Context, tpOpts *TransactionsPageOptions) (*TransactionsPage, error) {
+	return txs.getTransactionsByGroup(ctx, unconfirmed, tpOpts)
+}
+
+// returns an array of Transaction's for passed array of transaction ids or hashes
+func (txs *TransactionService) GetUnconfirmedTransactionsByIds(ctx context.Context, ids []string) ([]Transaction, error) {
+	return txs.getTransactionsByIds(ctx, unconfirmed, ids)
+}
+
+// returns partial Transaction by id or hash
+func (txs *TransactionService) GetPartialTransaction(ctx context.Context, id string) (Transaction, error) {
+	return txs.getTransaction(ctx, partial.String(), id)
+}
+
+// returns partial Transactions
+func (txs *TransactionService) GetPartialTransactions(ctx context.Context, tpOpts *TransactionsPageOptions) (*TransactionsPage, error) {
+	return txs.getTransactionsByGroup(ctx, partial, tpOpts)
+}
+
+// returns an array of Transaction's for passed array of transaction ids or hashes
+func (txs *TransactionService) GetPartialTransactionsByIds(ctx context.Context, ids []string) ([]Transaction, error) {
+	return txs.getTransactionsByIds(ctx, partial, ids)
+}
+
 // returns an array of Transaction's for passed array of transaction ids or hashes
 func (txs *TransactionService) GetTransactions(ctx context.Context, ids []string) ([]Transaction, error) {
```

Comment:
Maybe we don't need it, because you already added a new one?=)

Reply:
By my point of view a lot of small functions better =))

Reply:
From my point of view a lot of small functions better =))

Reply:
@Cempl so where are filters and paginations ?

Reply:
@Wondertan added [here](https://github.com/proximax-storage/go-xpx-chain-sdk/pull/85/commits/4072c37e8e379ac1458b31ccc30e97c313d9fde6) )